### PR TITLE
GitHub

### PR DESCRIPTION
**Pull request by Hoyg**
Modifying the lab

### CODE REVIEW

File: lab/readme.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+## Setting up a code repository on GitHub
+
+There are many ways to add code to a GitHub repository. We can start with a new empty repository and manually add files, commit and push. But for the purpose of this lab, we will ***Fork*** an existing repository. A *Fork* is a copy of a repository.  When you *Fork*, you are creating a personal copy of the repository - any changes that you make will not affect the original code. If you want  to contribute back to the original, you can submit a *Pull Request*
+
+1. Go to [**GitHub**](http://www.github.com) and make sure you are logged in with your GitHub user account.
+
+2.Navigate to the [**PartsUnlimitedE2E**](https://github.com/Microsoft/PartsUnlimitedE2E) repository and select **Fork** to your GitHub account.
+
+![Forking a GH repo](images/8.png)
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: Image is not aligned

File: lab/readme.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+   <img src="images/19.png">
+
+3. Save the build definition.
+
+
+## Continuous Integration 
+                
+
+
+We have an ASP.NET application code forked from the Microsoft GitHub site which will be deployed to Azure as a web app.
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: It should be "We have an ASP.NET application code forked from the Microsoft GitHub repo which will be deployed to Azure as as Web App"

File: lab/readme.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+3. Save the build definition.
+
+
+## Continuous Integration 
+                
+
+
+We have an ASP.NET application code forked from the Microsoft GitHub site which will be deployed to Azure as a web app. 
+
+To achieve this, we have to edit the code which will inturn trigger a CI build. Navigate to your forked GitHub repository.
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: We will achieve this by editing the code and running the CI build.

File: lab/readme.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+We have a Continuous Integration (CI) build setup to run upon a code commit. Let's make a simple change in the code to trigger the CI build.
+
+1. Open the file **Index.cshtml** under **master** branch by navigating to the below path-
+
+   >PartsUnlimitedE2E/PartsUnlimited-aspnet45/src/PartsUnlimitedWebsite/Views/Home/Index.cshtml
+
+<img src="images/13.png">
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: Image is not aligned

File: lab/readme.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+We have a Continuous Integration (CI) build setup to run upon a code commit. Let's make a simple change in the code to trigger the CI build.
+
+1. Open the file **Index.cshtml** under **master** branch by navigating to the below path-
+
+   >PartsUnlimitedE2E/PartsUnlimited-aspnet45/src/PartsUnlimitedWebsite/Views/Home/Index.cshtml
+
+<img src="images/13.png"> 
+
+<img src="images/14.png">
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: Image is not aligned

File: lab/readme.md [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+2. Edit the code. For this example, let's change **line 30** to increase discount from **50%** to **70%**.
+
+3. Navigate to **Preview changes**, enter a commit message and click on **Commit Changes**.
+
+   <img src="images/15.png"> 
+
+4. The code commit will trigger a CI build. Go to the **Build** tab within the project portal to see the CI build in progress.
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: It should be "Go to the build tab to see the in-progress build"

File: lab/readme.md [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+
+5. Once the build is completed, you can see the summary which shows **test results, code coverage** etc.
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: Once the build is complete, you will see the summary which shows **Test Results** and **Code Coverage**.

File: lab/readme.md [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+
+
+## Continuous Deployment
+
+Let us use Infrastructure as a Code in the release pipeline. We have a release configured to deploy the application which is associated to the build and triggered when the build is successful.
+
+  
+1. Go to Releases tab under Build and Release hub.
+
+<img src="images/30.png">
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: Image is not aligned.

File: lab/readme.md [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+
+2. Select the **GitHub** definition, you will see the release in-progress.
+
+3. While the release is in-progress, let's explore the tasks used. Click edit to see the release pipeline where the target environment is **Azure Deployment**.
+
+<img src="images/25.png">
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: Image is not aligned.

File: lab/readme.md [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+
+2. Select the **GitHub** definition, you will see the release in-progress.
+
+3. While the release is in-progress, let's explore the tasks used. Click edit to see the release pipeline where the target environment is **Azure Deployment**.
+
+<img src="images/25.png"> 
+
+4. Let us take a close look at the artifacts being used. The artifact used here is  **VSTS CI Team Build** artifact which is the resulting binaries of the CI build, is being used. Continuous Deployment condition has been enabled for this artifact.
+
+ <img src="images/26.png">
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: Image is not aligned.

File: lab/readme.md [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+
+5. Navigate to the **Azure Deployment** environment and you will see 2 tasks are used. Let us explore the tasks.
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: In the line 142 you have mentioned to explore tasks and why here as well? Let's us keep at 1 place and avoid confusions.

File: lab/readme.md [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+
+<img src="images/29.png"> 
+
+- Azure Resource Group Deployment: The project used in this lab has the **Azure App service** as its front-end and **Azure SQL DB service** as its back-end. We will provision these services as <a href="https://azure.microsoft.com/en-in/overview/what-is-paas/">PAAS</a> on Azure using <a href="https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-create-first-template">ARM templates</a>. This task will create the above services in a resource group GitHub.
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: What is **the above services** here? We should add a screenshot to the services which was created.

File: lab/readme.md [hunk 13]
```diff
@@ -0,0 +1,191 @@
...
+
+- Azure App Service Deploy: The task is used to deploy a Web project to the Azure App Service created above.
+
+<img src="images/33.png">
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: Image not aligned.

File: lab/readme.md [hunk 14]
```diff
@@ -0,0 +1,191 @@
...
+- Azure Resource Group Deployment: The project used in this lab has the **Azure App service** as its front-end and **Azure SQL DB service** as its back-end. We will provision these services as <a href="https://azure.microsoft.com/en-in/overview/what-is-paas/">PAAS</a> on Azure using <a href="https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-create-first-template">ARM templates</a>. This task will create the above services in a resource group GitHub.
+
+- Azure App Service Deploy: The task is used to deploy a Web project to the Azure App Service created above.
+
+<img src="images/33.png"> 
+
+
+7. Once the release is complete, you will see the task-wise summary.
+
+<img src="images/35.png">
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: Image not aligned.

File: lab/readme.md [hunk 15]
```diff
@@ -0,0 +1,191 @@
...
+- Azure App Service Deploy: The task is used to deploy a Web project to the Azure App Service created above.
+
+<img src="images/33.png"> 
+
+
+7. Once the release is complete, you will see the task-wise summary.
+
+<img src="images/35.png"> 
+
+<img src="images/36.png">
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: Image not aligned.

File: lab/readme.md [hunk 16]
```diff
@@ -0,0 +1,191 @@
...
+
+7. Once the release is complete, you will see the task-wise summary.
+
+<img src="images/35.png"> 
+
+<img src="images/36.png"> 
+
+8. Login to Azure Portal and search a Resource Group with the name **Github**.
+
+<img src="images/37.png">
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: Image not aligned.

File: lab/readme.md [hunk 17]
```diff
@@ -0,0 +1,191 @@
...
+
+9. Navigate to one of the Dev/Staging WebApps from the resource group and you will see the application deployed successfully with the changes.
+
+<img src="images/38.png">
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: Image not aligned.

File: lab/readme.md [hunk 18]
```diff
@@ -0,0 +1,191 @@
...
+
+<img src="images/38.png"> 
+
+## Summary
+
+Whether your code lives in **Visual Studio Team Services** or **GitHub** (or both), manage build, test, and deploy in one place. Create and manage a Continuous Integration process that connects to GitHub, gets the source files, compiles and deploys the binaries to any environment with Continuous Deployment feature.
+
+## Feedback
+
+if you have any feedback, write to <a href="mailto: devopsdemos@microsoft.com" target=_blank>us</a>
```
Comment (SrivatsaMarichi) [inline lab/readme.md]: It should be "Please let <a href="mailto: devopsdemos@microsoft.com" target=_blank>us </a>know if you have any feedback on this lab."

File: lab/readme.md (hunk 1 above)
Comment (Hoyg) [reply lab/readme.md]: Done

File: lab/readme.md (hunk 8 above)
Comment (Hoyg) [reply lab/readme.md]: done

**Pull request merged**